ActiveLabeler-main/ActiveLabeler.py

```diff
@@ -0,0 +1,181 @@
+import random
+from tqdm import tqdm
+import numpy as np
+import torch
+from torch.utils.data import DataLoader
+from torchvision import transforms
+from torchvision.datasets import ImageFolder
+
+class ActiveLabeler:
+    """Active labeler code which is used with a SSL model to get curated dataset from unlabeled data"""
+
+    def __init__(self, embeddings, images_path,image_size,batch_size,seed):
+        """Initialise with embeddings and respected image paths"""
+        self.embeddings = embeddings
+        self.images_path = images_path
+        self.image_size = image_size
+        self.batch_size = batch_size
+        self.probabilities = []
+        np.random.seed(seed)
+        random.seed(seed)
+
+    def strategy(self, name, query, N, strategy_params):
+        """
+        Get a subset of unlabled images picked by a specific strategy
+
+        Keyword arguments
+        name -- Name of the sampling strategy to use
+                Options: uncertainty, uncertainty_balanced, random, positive, gaussian
+
+        query -- A num_images X num_classes sized numpy array. This numpy array has the softmax probability
+                    of each image. These softmax probs are predicted by the latest finetune model. Type: numpy array
+
+        N -- Number of images that should be in the subset. Type: integer
+
+        strategy_params -- A dictionary of strategy specific args. Type: Dictionary
+
+        Returns list/numpy array of the index of images in the subset.
+        """
+        if name == "gaussian":
+
+            def gaussian(x, mean, sigma):
+                a = 1 / (sigma * np.sqrt(2 * np.pi))
+                exp_val = -(1 / 2) * (((x - mean) ** 2) / (sigma ** 2))
+                tmp = list(a * np.exp(exp_val))[0]
+                return tmp
+
+            def get_gaussian_weights(confidences):
+                mean = 0.5
+                sigma = 0.2
+                return [gaussian(conf_i, mean, sigma) for conf_i in confidences]
+
+            def get_samples(image_list, confidences, number_of_samples):
```

Comment:
Nitpick: `number_of_samples` could be renamed as `sample_count` to avoid the need for a preposition.